django/db/models/lookups.py

```diff
@@ -55,6 +55,17 @@ def batch_process_rhs(self, compiler, connection, rhs=None):
             sqls, sqls_params = ['%s'] * len(params), params
         return sqls, sqls_params
 
+    def get_source_expressions(self):
+        if self.rhs_is_direct_value():
+            return [self.lhs]
+        return [self.lhs, self.rhs]
```

Comment:
I'd love to see some comments here and the set_source_expressions method below that tells us why this is necessary.